### PR TITLE
Searchgame

### PR DESCRIPTION
**Pull request by katemorris**
### Description

Connects the backend and Sinatra. The URL for Sinatra was added to Travis CI so it can run successfully as well. This feature allows a search term to be fed in. If the game with that exact name string exists in the DB, it returns that game. If it does not exist, the sinatra service is queried. The return values are passed back in JSON. 

This adds VCR and Webmock gems as well, so please bundle install when you run things next!

### Closes issue(s)

#4 

### Testing

All tests done for facades, services, and happy/sad paths.

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [X] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme (coming after it is live on Heroku and we can query postman)
- [X] I ran rubocop and fixed issues
- [X] I ran full test suite - all tests passing 

### Other comments
Also merged with current main and resolved merge conflicts

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -5,4 +5,9 @@ class Game < ApplicationRecord
   validates :age_range, presence: true
   validates :duration, presence: true
   validates :num_players, presence: true
+
+  def self.find_game_db(game_name)
+    # KT - changed to be exact so that if it's not exactly what they searched for, it triggers the API search
```
[the remark this answers is not in the document]
Comment (katemorris) [reply app/models/game.rb:10]: It's a comment, go away.

File: spec/requests/api/v1/game_search/request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                         image: 'Knight',
+                         num_players: '2',
+                         age_range: 6)
+
+    get "/api/v1/games/find?name=#{chess.name}"
+
+    expect(response.status).to eq(200)
+
+    games = JSON.parse(response.body, symbolize_names: true)[:data]
+    games.each do |games|
```
Comment (katemorris) [inline spec/requests/api/v1/game_search/request_spec.rb]: Updating!

File: app/services/search_service.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class SearchService
+  def self.get_games_list_by_name(query)
+    response = conn.get("/api/v1/search?name=#{query}").body
```
Comment (katemorris) [inline app/services/search_service.rb]: Valid

File: db/seeds.rb [hunk 4]
```diff
@@ -6,11 +6,40 @@
...
+                   image: '',
+                   google_token: 'asdf34qasdf',
+                   refresh_token: '21354q12efraswdf',
+                   auth_token: 'q234rdf',
+                   email: 'Phil@example.com')
+
+Friend.create(user_id: kate.id, bud_id: austin.id)
+Friend.create(user_id: kate.id, bud_id: phil.id)
+
+chess = Game.create(name: 'Chess',
```
Comment (katemorris) [inline db/seeds.rb]: Okay

**Review by ShaunDaneJames — APPROVED** [review]
LGTM!

**Review by PhilipDeFraties — APPROVED** [review]

**Pull request merged**